tests/spki_tests.rs

```diff
@@ -0,0 +1,700 @@
+///
+/// Allows signature verification with various flavors of right and wrong parameters: using
+/// one key to sign but comparing against another key's signature, etc.
+///
+/// This loads keys, input files, signatures, etc from the hierarchy in tests/test-data/.
+///
+/// algorithm: spki verification algorithm
+/// key_id: the key ("rsa_2048", "ecdsa_secp256k1", etc) when loading the key pair
+/// key_sig_id: the key when loading a pre-calculated signature file. If different from key_id,
+///     verification will fail, of course.
+/// digest_alg: digest alg when loading a signature file.  If it doesn't match the digest in
+///     `algorithm`, verification will fail.
+/// message_id: the input file to verify the signature of. Either "msg1" or "msg2".
+/// msg_sig_id: the input file to load the pre-existing signature of. If different from
+///     `message_id`, verification will fail.
+fn ec_verify_signature_file_with_spki(algorithm: &spki::Algorithm,
+                                      key_id: &str,
+                                      key_sig_id: &str,
+                                      digest_alg: &str,
+                                      message_id: &str,
+                                      msg_sig_id: &str,
+                                      expected: Result<(), VerifyWithSPKIError>) {
+    verify_signature_with_spki(algorithm,
+                               Path::new(&format!("tests/test-data/messages/{}.bin", message_id)),
+                               Path::new(&format!("tests/test-data/signatures/{}_{}_{}_sig.bin",
+                                                  msg_sig_id,
+                                                  key_sig_id,
+                                                  digest_alg)),
+                               Path::new(&format!("tests/test-data/keys/{}_pub_spki.der", key_id)),
+                               expected);
+}
+
+fn verify_signature_with_spki(algorithm: &spki::Algorithm,
+                              message_path: &Path,
+                              sig_path: &Path,
+                              spki_path: &Path,
+                              expected: Result<(), VerifyWithSPKIError>) {
+    let input = read_file_completely(message_path);
+    let signature = read_file_completely(sig_path);
+
+    let spki_bytes = read_file_completely(spki_path);
+    let spki_input = untrusted::Input::from(&spki_bytes);
+
+    // Verify the signature.
+    assert_eq!(expected, spki::verify(algorithm,
+                                      spki_input,
+                                      untrusted::Input::from(&input),
+                                      untrusted::Input::from(&signature)));
+}
+
+fn read_file_completely(path: &Path) -> Vec<u8> {
+    let mut f = File::open(path).unwrap();
+    let mut vec = Vec::new();
+    assert!(f.read_to_end(&mut vec).unwrap() > 0);
+    return vec;
+}
+
+// TODO sha1 support?
```

Comment:
We do support verifying RSA PKCS#1 SHA-1 signatures but not RSA PSS SHA-1 signatures. We don't support *signing* with SHA-1 at all.